diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,5 +1,11 @@
 load("@rules_cc//cc:defs.bzl", "cc_library")

+filegroup(
+    name = "all_files",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "darts-clone",
     hdrs = glob(["include/*.h"]),
diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,4 +1,4 @@
-"""Darts-clone is a clone of Darts (Double-ARray Trie System) which is a C++ header library for a static double-array trie structure."""
+"""Darts-clone is a clone of Darts (Double-ARray Trie System)."""

 module(
     name = "darts-clone",
@@ -6,4 +6,4 @@
     compatibility_level = 0,
 )

-bazel_dep(name = "rules_cc", version = "0.2.17")
+bazel_dep(name = "rules_cc", version = "0.2.20")
diff --git a/include/darts.h b/include/darts.h
--- a/include/darts.h
+++ b/include/darts.h
@@ -1,7 +1,9 @@
 #ifndef DARTS_H_
 #define DARTS_H_

+#include <cstdint>
 #include <cstdio>
+#include <cstring>
 #include <exception>
 #include <new>

@@ -36,7 +38,7 @@

 // The main structure of Darts-clone is an array of <DoubleArrayUnit>s, and the
 // unit type is actually a wrapper of <id_type>.
-typedef unsigned int id_type;
+typedef uint32_t id_type;

 // <progress_func_type> is the type of callback functions for reporting the
 // progress of building a dictionary. See also build() of <DoubleArray>.
@@ -172,6 +174,14 @@
     clear();
     array_ = static_cast<const unit_type *>(ptr);
     size_ = size;
+  }
+  void copy_array(const void* ptr, std::size_t num_bytes) {
+    clear();
+    const std::size_t extra_padding = num_bytes % unit_size() == 0 ? 0 : 1;
+    size_ = num_bytes / unit_size() + extra_padding;
+    buf_ = new unit_type[size_];
+    std::memcpy(buf_, ptr, num_bytes);
+    array_ = buf_;
   }
   // array() returns a pointer to the array of units.
   const void *array() const {
@@ -314,6 +324,8 @@
   // updates `node_pos' and `key_pos' after each transition.
   inline value_type traverse(const key_type *key, std::size_t &node_pos,
       std::size_t &key_pos, std::size_t length = 0) const;
+
+  bool validate(value_type max_value_limit = -1) const;

  private:
   typedef Details::uchar_type uchar_type;
@@ -454,6 +466,29 @@
 }

 template <typename A, typename B, typename T, typename C>
+bool DoubleArrayImpl<A, B, T, C>::validate(value_type max_value_limit) const {
+  if (size_ == 0 || array_ == NULL) {
+    return false;
+  }
+  if (array_[0].label() != '\0' || array_[0].has_leaf() ||
+      array_[0].offset() == 0 || ((0 ^ array_[0].offset()) | 0xFF) >= size_) {
+    return false;
+  }
+  for (std::size_t i = 1; i < size_; ++i) {
+    if (array_[i].label() <= 0xFF) {
+      if (((i ^ array_[i].offset()) | 0xFF) >= size_) {
+        return false;
+      }
+    } else if (max_value_limit >= 0) {
+      if (array_[i].value() >= max_value_limit) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+template <typename A, typename B, typename T, typename C>
 template <typename U>
 inline U DoubleArrayImpl<A, B, T, C>::exactMatchSearch(const key_type *key,
     std::size_t length, std::size_t node_pos) const {
