diff --git a/node_addon/defs.bzl b/node_addon/defs.bzl
index 1111111..2222222 100644
--- a/node_addon/defs.bzl
+++ b/node_addon/defs.bzl
@@ -3,8 +3,6 @@
 load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
 load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
 load("@rules_cc//cc:defs.bzl", "cc_binary")
-load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
-load("@rules_cc//cc/private:cc_common.bzl", "cc_common")
 
 _NODE_ADDON_API = Label("@node_addon_api//:node_addon_api")
 _NODE_ADDON_API_HEADERS_ONLY = Label("@node_addon_api//:node_addon_api_headers_only")
