From: Shengjing Zhu <zhsj@debian.org>
Date: Fri, 2 Aug 2024 17:11:08 +0800
Subject: Disable build in setup.py

Last-Update: 2026-07-04
---
--- a/setup.py	2026-07-04 01:16:39.397276721 +0000
+++ b/setup.py	2026-07-04 01:17:17.094201906 +0000
@@ -98,6 +98,7 @@
 
 
 def build_libopencc(output_path):
+    return
     print('building libopencc into %s' % _build_dir)
 
     is_windows = sys.platform == 'win32'
